pipeline/load.py

```diff
@@ -0,0 +1,343 @@
+                    VALUES (%s, %s, %s, %s, %s, %s);
+                """, (latitude, longitude, town, country_code, continent_id, city))
+                print(
+                    f"Inserted: ({latitude}, {longitude}, {town}, {country_code}, {continent_id}, {city})")
+            else:
+                print(f"""Duplicate: {latitude}, {longitude}, {
+                      town}, {country_code}, {continent_id}, {city}""")
+
+
+def insert_plants(conn: 'connection', plant_df: pd.DataFrame) -> None:
+    """Inserts plant information into the database
+        Ignores duplicate entries"""
+    with conn.cursor() as cur:
+        for index, row in plant_df.iterrows():
+            plant_id = row["plant_id"]
+            plant_name = row["name"]
+            scientific_name = row["scientific_name"]
+            latitude = row["Latitude"]
+            longitude = row["Longitude"]
+            image_url = row["images.original_url"]
+            town = row["Town"]
+            country_code = row["Country_Code"]
+            continent = row["Continent"]
+            city = row["City"]
+
+            print(plant_id)
+
+            try:
+                cur.execute("""SELECT continent_id
+                            FROM delta.Continents
+                            WHERE continent_name = %s;
+                            """, (continent))
+                continent_id = cur.fetchone()[0]
+            except:
+                continue
+            if pd.isna(scientific_name):
+                scientific_name = "None"
+            scientific_name_id = find_scientific_name_id(
+                conn, scientific_name)
+
+            if pd.isna(image_url):
+                image_url = "None"
+
+            location_id = find_location_id(
+                conn, latitude, longitude, town, country_code, continent_id, city)
+
+            cur.execute("""
+                SELECT COUNT(*)
+                FROM delta.Plants
+                WHERE plant_id = %s AND plant_name = %s AND scientific_id = %s AND location_id = %s
+                        AND image_url = %s;
+            """, (plant_id, plant_name, scientific_name_id, location_id, image_url))
+
+            count = cur.fetchone()[0]
+
+            if count == 0:
+                cur.execute("""
+                    INSERT INTO delta.Plants (plant_id, plant_name, scientific_id, location_id, image_url)
+                    VALUES (%s, %s, %s, %s, %s);
+                """, (plant_id, plant_name, scientific_name_id, location_id, image_url))
+                print(
+                    f"Inserted: ({plant_id}, {plant_name}, {scientific_name_id}, {location_id}, {image_url})")
+            else:
+                print(f"""Duplicate: ({plant_id}, {plant_name}, {
+                    scientific_name_id}, {location_id}, {image_url})""")
+
+
+def insert_recording(conn: 'connection', plant_df: pd.DataFrame) -> None:
+    with conn.cursor() as cur:
```

Comment:
instead of having mutliple `with conn.cursor() as cur`, can we pass in a cursor variable instead?